audit.go

```diff
@@ -0,0 +1,60 @@
+package authaus
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/IMQS/serviceauth"
+)
+
+type Auditor interface {
+	AuditUserAction(identity, password, actionDescription string)
+}
+
+type Action struct {
+	Who     string         `json:"who"`
+	DidWhat string         `json:"did_what"`
+	ToWhat  string         `json:"to_what"`
+	AtTime  int64          `json:"at_time"`
+	Context *ActionContext `json:"context"`
+}
+
+type ActionContext struct {
+	Location string `json:"location"`
+}
+
+func (x *Central) AuditUserAction(identity, password, actionDescription string) {
+	action := &Action{}
+	action.Who = identity
+	action.DidWhat = actionDescription
+	action.ToWhat = "IMQS Web V8"
+	action.AtTime = time.Now().Unix()
+
+	actionContext := &ActionContext{}
+	actionContext.Location = time.Now().Location().String()
+	action.Context = actionContext
+
+	client := &http.Client{}
+	actionStr, _ := json.Marshal(action)
+
+	req, err := http.NewRequest("POST", x.AuditServiceUrl, bytes.NewBuffer(actionStr))
+	if err != nil {
+		x.Log.Infof("Error creating audit request: (%v)", err)
```

Comment:
If you have an error here (or in subsequent places in this function), you cannot just continue executing as though there is no error. If NewRequest returns an error, then you can expect 'req' to be null, so you would just end up crashing on the next line. Also, this is not log level Info, this is Error.